src/components/flame-graph/Canvas.js

```diff
@@ -199,10 +199,10 @@ class FlameGraphCanvas extends React.PureComponent<Props> {
         const startTime = stackTiming.start[i];
         const endTime = stackTiming.end[i];
 
-        const x: CssPixels = startTime * containerWidth;
+        const x: CssPixels = startTime + containerWidth;
         const y: CssPixels =
-          (maxStackDepth - depth - 1) * ROW_HEIGHT - viewportTop;
-        const w: CssPixels = (endTime - startTime) * containerWidth;
+          (maxStackDepth + depth + 1) * ROW_HEIGHT - viewportTop;
+        const w: CssPixels = (endTime + startTime) * containerWidth;
```

Comment:
Why did you have to change the computations of these x, y and w values? I don't think this is correct.

Reply:
OK, @canova  I am working on it.

Reply:
Hello, @canova I had worked on my tasks and had corrected tests only one yarn test was left to solve when I push my code I am seeing lots of correction in yarn flow, lint, tests, and those files did not work also.  These are not my work:
![eer](https://user-images.githubusercontent.com/57604500/96922405-17248100-14b0-11eb-86d6-abfa73fb17b2.png)

Reply:
I don't see these errors anymore but I see this:
```
Error ------------------------------------------------------------------ src/components/flame-graph/FlameGraph.js:288:11

Cannot create `FlameGraphCanvas` element because `ThunkAction` [1] is incompatible with undefined [2] in the return
value of property `chartProps.updatePreviewSelection`.

   src/components/flame-graph/FlameGraph.js:288:11
                   v----------------
    288|           <FlameGraphCanvas
    289|             key={threadsKey}
    290|             // ChartViewport props
    291|             viewportProps={{
    292|               timeRange,
    293|               maxViewportHeight,
    294|               maximumZoom: 1,
    295|               previewSelection,
    296|               startsAtBottom: true,
    297|               disableHorizontalMovement: true,
    298|               viewportNeedsUpdate,
    299|               marginLeft: 0,
    300|               marginRight: 0,
    301|               containerRef: this._takeViewportRef,
    302|             }}
       :
    315|               categories,
    316|               selectedCallNodeIndex,
    317|               scrollToSelectionGeneration,
    318|               callTreeSummaryStrategy,
    319|               stackFrameHeight: STACK_FRAME_HEIGHT,
    320|               onSelectionChange: this._onSelectedCallNodeChange,
    321|               onRightClick: this._onRightClickedCallNodeChange,
    322|               shouldDisplayTooltips: this._shouldDisplayTooltips,
    323|               interval,
    324|               isInverted,
    325|               samples,
    326|               unfilteredSamples,
    327|               tracedTiming,
    328|             }}
    329|           />
                   -^

References:
   src/actions/profile-view.js:1277:4
   1277| ): ThunkAction<void> {
            ^^^^^^^^^^^^^^^^^ [1]
   src/actions/profile-view.js:1277:16
   1277| ): ThunkAction<void> {
                        ^^^^ [2]
```
That's because you added a new `updatePreviewSelection` prop but didn't send that prop to the component where we render this canvas component. You also need to pass this prop so you can use it there.
Also good to note that I didn't check if this is the right solution. I only looked at the flow error.